packages/core/types/index.js

```diff
@@ -173,18 +174,23 @@ export type InitialParcelOptions = {|
   // detailedReport
 |};
 
-export type ParcelOptions = {|
-  ...InitialParcelOptions,
-  cacheDir: FilePath,
-  entries: Array<FilePath>,
-  logLevel: LogLevel,
-  rootDir: FilePath,
-  targets: Array<Target>,
-  projectRoot: FilePath,
-  lockFile: ?FilePath,
-  inputFS: FileSystem,
-  outputFS: FileSystem
-|};
+export interface PluginOptions {
```

Comment:
For a situation like this I kind of prefer the exact object type over the interface, so that if someone typos a property name it causes a type error. We can still accomplish readonly properties like this:

```
export type ParcelOptions = {|
  +cacheDir: FilePath,
  +entries: Array<FilePath>,
  +logLevel: LogLevel,
  ...
|};
```

or even with `$ReadOnly`[0]:

```
export type ParcelOptions = $ReadOnly<{|
  cacheDir: FilePath,
  entries: Array<FilePath>,
  logLevel: LogLevel,
  ...
|}>;
```

What was the rationale behind using an interface?

[0] https://flow.org/en/docs/types/utilities/#toc-readonly